### PR TITLE
[CSI] Update images to latest

### PR DESCRIPTION
**Pull request by ggriffiths**
Signed-off-by: Grant Griffiths <ggriffiths@purestorage.com>

**What this PR does / why we need it**:

**Which issue(s) this PR fixes** (optional)
Closes #

**Special notes for your reviewer**:

### CODE REVIEW

**Review by haibinxie — COMMENTED** [review]
Is there any functional test automation? what's the process to verify if the new images wont introduce regression

**Comment by ggriffiths** [comment]
We have automated functional testing with the csi-e2e-test suite. Also torpedo CSI tests cover this path.

File: drivers/storage/portworx/component/csi.go [hunk 1]
```diff
@@ -563,6 +563,7 @@ func getCSIDeploymentSpec(
 			"--csi-address=$(ADDRESS)",
 			"--leader-election=true",
 			"--default-fstype=ext4",
+			"--extra-create-metadata=true",
```
Comment (pure-jliao) [inline drivers/storage/portworx/component/csi.go:566]: provisioner v2+ is not covered in unit tests, we may want to add that as the default in the future

File: test/integration_test/testspec/migration/daemonset-with-all-components.yaml [hunk 2]
```diff
@@ -132,7 +132,7 @@ spec:
             - name: dbusmount
               mountPath: /var/run/dbus
         - name: csi-node-driver-registrar
-          image: k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.1.0
+          image: k8s.gcr.io/sig-storage/csi-node-driver-registrar:v2.5.0
```
Comment (pure-jliao) [inline test/integration_test/testspec/migration/daemonset-with-all-components.yaml:135]: do we update the default versions for daemonset installation? if not probably we don't need to change here.

**Review by pure-jliao — APPROVED** [review]
LGTM

File: test/integration_test/testspec/migration/daemonset-with-all-components.yaml (hunk 2 above)
Comment (ggriffiths) [reply test/integration_test/testspec/migration/daemonset-with-all-components.yaml:135]: Yes we updated the version for the daemonset version

File: drivers/storage/portworx/component/csi.go (hunk 1 above)
Comment (ggriffiths) [reply drivers/storage/portworx/component/csi.go:566]: Ok yes we can add that in the future

**Pull request merged**